src/commands/new.rs

```diff
@@ -1,27 +1,75 @@
-use anyhow::Result;
-use spin_templates::TemplatesManager;
-use std::path::PathBuf;
+use anyhow::{anyhow, Context, Result};
+use spin_templates::{RunOptions, TemplateManager};
+use std::{collections::HashMap, path::PathBuf, str::FromStr};
 use structopt::StructOpt;
 
-/// Scaffold a new application locally based on a template.
+/// Scaffold a new application or component based on a template.
 #[derive(StructOpt, Debug)]
 pub struct NewCommand {
-    /// The local templates repository.
-    #[structopt(long = "repo")]
-    pub repo: String,
+    /// The template from which to create the new application or component.
+    pub template_id: String,
 
-    /// The name of the template.
-    #[structopt(long = "template")]
-    pub template: String,
+    /// The name of the new application or component.
+    #[structopt(short = "n", long = "name")]
+    pub name: Option<String>,
 
-    /// The destination where the template will be used.
-    #[structopt(long = "path")]
-    pub path: PathBuf,
+    /// The directory in which to create the new application or component.
+    #[structopt(short = "o", long = "output")]
+    pub output_path: Option<PathBuf>,
+
+    /// Parameter values to be passed to the template (in name=value format).
+    #[structopt(short = "v", long = "value", multiple = true)]
+    pub values: Vec<ParameterValue>,
 }
 
 impl NewCommand {
-    pub async fn run(self) -> Result<()> {
-        let tm = TemplatesManager::default().await?;
-        tm.generate(&self.repo, &self.template, self.path).await
+    pub async fn run(&self) -> Result<()> {
+        let template_manager =
+            TemplateManager::default().context("Failed to construct template directory path")?;
+        let template = template_manager
+            .get(&self.template_id)
+            .with_context(|| format!("Error retrieving template {}", self.template_id))?;
+        let options = RunOptions {
+            name: self.name.clone(),
+            output_path: self.output_path.clone(),
+            values: to_hash_map(&self.values),
+        };
+
+        match template {
+            Some(template) => template.run(options).interactive().execute().await,
+            None => {
+                // TODO: guidance experience
+                println!("Template {} not found", self.template_id);
+                Ok(())
+            }
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct ParameterValue {
+    pub name: String,
+    pub value: String,
+}
+
+impl FromStr for ParameterValue {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        if let Some((name, value)) = s.split_once('=') {
+            Ok(Self {
+                name: name.to_owned(),
+                value: value.to_owned(),
+            })
+        } else {
+            Err(anyhow!("'{}' should be in the form name=value", s))
+        }
     }
 }
+
+fn to_hash_map(values: &[ParameterValue]) -> HashMap<String, String> {
```

Comment:
minor: Could be `impl Into<HashMap<String, String>> for &[ParameterValue]`?

Reply:
![image](https://user-images.githubusercontent.com/865538/165629541-1f8fe745-7d1f-4cd2-911d-1debebc7ace1.png)

Reply:
I have a feeling I tried that before and was thwarted in this exact way!

Reply:
Ah yeah, I was thinking you could implement for `&[LocalType]`, but I guess it makes sense for coherence that you can't.